model/Concept.php

```diff
@@ -606,8 +606,11 @@ public function getProperties()
                             // checking if the property value is not in the current vocabulary
                             $exvoc = $this->model->guessVocabularyFromURI($val->getUri(), $this->vocab->getId());
                             if ($exvoc && $exvoc->getId() !== $this->vocab->getId()) {
-                                $value = new ConceptPropertyValue($this->model, $exvoc, $val, $prop, $this->clang, true);
+                                $voc = $exvoc;
+                            } else {
+                                $voc = $this->vocab;
                             }
+                            $value = new ConceptPropertyValue($this->model, $voc, $val, $prop, $this->clang, true);
```

Comment:
Now this object is created twice in case of `else` clause. It might be nicer and more efficient to just set the `external` value of ConceptPropertyValue.

Reply:
The `external` value is not the only thing that changes. Also the `vocab` parameter (second parameter) is different in the second constructor call.

But I agree this is not very clean, I will try to implement another approach.

Reply:
I've now moved this logic inside the ConceptPropertyValue constructor.